### ISSUE TITLE
Update "form" explaination in Glossary

### ISSUE THREAD

**Issue by akshay-ap** [opendatakit/docs#605]
Through recent conversation, it seems like there can be a more clear explanation for the word `form`.

https://opendatakit.slack.com/archives/C5KCRU1L1/p1521481777000547

In current docs, the explanation of a `form` in the [glossary](https://docs.opendatakit.org/glossary/) is stated as `A defined set of questions and answer choices displayed by an application that can render forms written in the XForm standard.`. 
According to me, the correct word should be `blank form`.
I propose to add the explanation as follows:

 `form`: An XML document written in the XForm standard.

- `blank form`: Stores information about the type of questions, constraints on answers and options (if any) for each question which is rendered by the application. It is generally referred just as `form` in the documentation.

- `filled form`: Stores the user responses. A filled form may also contain other files like images, audio, video.

**Comment by adammichaelwood** [comment]
> form: An XML document written in the XForm standard.

I like this clarity, and the addition of two additional glossary terms. Your wording on `blank` and `filled` could be a little clearer.

The only thing I'm uncomfortable about here is that the clarity (form=XML document) eliminates the *idea* of a form, and the rendered form. I think of "form" in the abstract, with several manifestations: form definition (blank form), a filled in form, a rendered form (in app)

**Comment by akshay-ap** [comment]
Yes, form = XML document won't be a right explanation. Let's see what other community members have to say about it and the terms to be used for explaining `blank` and `filled` forms.

**Comment by akshay-ap** [comment]
I just noticed that current explanation for `form` is recursive:

**form**
A defined set of questions and answer choices displayed by an application that can render **forms** written in the XForm standard. Isn't it?

**Comment by adammichaelwood** [comment]
True. Also, I wrote that when I was still very new to the project. I imagine a lot of the glossary terms need some re-thinking.
